lib/ansible/modules/cloud/openstack/os_keystone_endpoint.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/python
+# Copyright (c) 2017 VEXXHOST, Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: os_keystone_endpoint
+short_description: Manage OpenStack Identity service endpoints.
+extends_documentation_fragment: openstack
+author: "Mohammed Naser (@mnaser)"
+version_added: "2.4"
+description:
+    - Create, update, or delete OpenStack Identity service endpoints. If a
+      service with the same combination of I(service), I(interface) and I(region)
+      exist, the I(url) and I(state) (C(present) or C(absent)) will be updated.
+options:
+   service:
+     description:
+        - Name of the service.
+     required: true
+   interface:
+     description:
+        - Interface of the service.
+     choices: [admin, public, internal]
+     required: true
+   url:
+     description:
+        - URL of the service.
+     required: true
+   region:
+     description:
+        - Region that the service belongs to. Note that I(region_name) is used for authentication.
+     required: true
+   enabled:
+     description:
+        - Is the service enabled.
+     default: True
+   state:
+     description:
+       - Should the resource be C(present) or C(absent).
+     choices: [present, absent]
+     default: present
+requirements:
+    - "python >= 2.6"
+    - "shade"
+'''
+
+EXAMPLES = '''
+# Create a service for glance
+- os_keystone_endpoint:
+     cloud: mycloud
+     state: present
+     service: glance
+     interface: public
+     url: http://controller:9292
+     region: RegionOne
+
+# Delete a service
+- os_keystone_endpoint:
+     cloud: mycloud
+     state: absent
+     name: image
+     interface: public
+     region: RegionOne
+'''
+
+RETURN = '''
+endpoint:
+    description: Dictionary describing the endpoint.
+    returned: On success when I(state) is C(present)
+    type: complex
+    contains:
+        id:
+            description: Endpoint ID.
+            type: string
+            sample: "3292f020780b4d5baf27ff7e1d224c44"
+        region:
+            description: Region Name.
+            type: string
+            sample: "RegionOne"
+        service_id:
```

Comment:
this isn't correct according to the code below, should be 'service' and accept name_or_id

Reply:
`service_id` is actually correct, this is the info returned.

```
TASK [debug] **********************************************************************************************************************************************************************
ok: [controller.example.com] => {
    "msg": {
        "changed": true,
        "msg": "All items completed",
        "results": [
            {
                "_ansible_item_result": true,
                "_ansible_no_log": false,
                "_ansible_parsed": true,
                "changed": true,
                "endpoint": {
                    "HUMAN_ID": false,
                    "NAME_ATTR": "name",
                    "enabled": true,
                    "human_id": null,
                    "id": "34ab50cb74834ec8911709220eaa92af",
                    "interface": "admin",
                    "links": {
                        "self": "http://controller.example.com:5000/v3/endpoints/34ab50cb74834ec8911709220eaa92af"
                    },
                    "region": "RegionOne",
                    "region_id": "RegionOne",
                    "service_id": "4e26c3decfdb4d37907fc798b2828fd4",
                    "url": "http://controller.example.com:9292"
                },
                "invocation": {
                    "module_args": {
                        "api_timeout": null,
                        "auth": null,
                        "auth_type": null,
                        "availability_zone": null,
                        "cacert": null,
                        "cert": null,
                        "cloud": "default",
                        "enabled": true,
                        "endpoint_type": "public",
                        "interface": "admin",
                        "key": null,
                        "region": "RegionOne",
                        "region_name": null,
                        "service": "glance",
                        "state": "present",
                        "timeout": 180,
                        "url": "http://controller.example.com:9292",
                        "verify": null,
                        "wait": true
                    }
                },
                "item": [
                    {
                        "service": "glance"
                    },
                    {
                        "interface": "admin",
                        "url": "http://controller.example.com:9292"
                    }
                ]
            }
        ]
    }
}
```

Reply:
I think you mean in line 23 should accept service's name or id as input.

Reply:
Fixed